cli/Dockerfile

```diff
@@ -20,13 +20,6 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      && apt-get update && apt-get install -y gh \ 
      # az cli
      && curl -sSL https://aka.ms/InstallAzureCLIDeb | bash \
-     # bicep
-     # Fetch the latest Bicep CLI binary
-     && curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64 \
-     # Mark it as executable
-     && chmod +x ./bicep \
-     # Add bicep to your PATH (requires admin)
-     && mv ./bicep /usr/local/bin/bicep \
      # nodejs
```

Comment:
Maybe we should consider installing this in our CI docker container image. I would hate to see CI builds starting to fail because we've introduced a ~40MB installation on each run.

I do wonder if respecting the copy installed in `$PATH` might be the way to go for CI scenarios.

Reply:
> I would hate to see CI builds starting to fail because we've introduced a ~40MB installation on each run.

Good call, I would hate this as well.

> I do wonder if respecting the copy installed in $PATH might be the way to go for CI scenarios.

I didn't want to walk down the `$PATH` route just yet (`az` didn't do it, for example).  We do support setting an explicit path, via config, so I'll do that here, after downloading the bicep CLI during the container build.

Reply:
This is the container that is used for our CI runs.  Are we sure we want to remove this?

Reply:
I'm a little torn on supporting `PATH` or this explicit override. Going for the override makes sense until we hear community feedback otherwise, but it definitely creates friction of how to set up `azd` correctly in CI for success. Given that we mostly just recommend using the container and have that control, I think this is fine for now.